src/data_type/mod.rs

```diff
@@ -2400,6 +2400,33 @@ impl DataType {
             _ => panic!(),
         }
     }
+
+    /// Return the empty datatype of the same variant
+    pub fn empty(&self) -> DataType {
```

Comment:
Call this `try_empty -> Result<DataType>`

Because this does not always work.

If it was always working it should be in the `Variant` trait